cmd/authorization/main.go

```diff
@@ -74,9 +93,119 @@ func uppercaseHandler(w http.ResponseWriter, req *http.Request) {
 	return
 }
 
+func createUser(db *sql.DB, login, password_hash string) error {
+	createUserStmt := `INSERT INTO users VALUES($1, $2)`
+	_, err := db.Exec(createUserStmt, login, password_hash)
+
+	if err != nil {
+		log.Println("Error create user " + err.Error())
+		return err
+	}
+	return err
+}
+func readUser(db *sql.DB, login string) (User, error) {
+	var user User
+	readUserStmt := `SELECT * FROM users WHERE login = $1`
+
+	rows, err := db.Query(readUserStmt, login)
+	if err != nil {
+		log.Println("Error get user " + err.Error())
+		return user, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		err = rows.Scan(&user)
+		if err != nil {
+			log.Println("Error get data from row " + err.Error())
+			return user, err
+		}
+	}
+	return user, nil
+}
+
+func updateUser(db *sql.DB, login, password_hash string) error {
+	updateUserStmt := `UPDATE users SET password_hash = $1 WHERE login = $2`
+	_, err := db.Exec(updateUserStmt, password_hash, login)
+
+	if err != nil {
+		log.Println("Error update user " + err.Error())
+		return err
+	}
+	return err
+}
+
+func deleteUser(db *sql.DB, login string) error {
+	deleteUserStmt := `DELETE FROM users WHERE login = $1`
+
+	_, err := db.Exec(deleteUserStmt, login)
+
+	if err != nil {
+		log.Println("Error delete user " + err.Error())
+		return err
+	}
+
+	return err
+}
+
+func createToken(db *sql.DB, token string, userID int64, expired time.Time) error {
+	createTokenStmt := `INSERT INTO tokens (token, user_id) VALUES($1, $2)`
```

Comment:
You need to check if such userID exists.

Reply:
so there will be two queries? First one for check if current id exists and the second one to create token?

Reply:
Correct.